doc/manual/src/language/constructs.md

```diff
@@ -2,8 +2,11 @@
 
 ## Recursive sets
 
-Recursive sets are just normal sets, but the attributes can refer to
-each other. For example,
+Recursive sets are like normal [attribute sets](./values.md#attribute-set), but the attributes can refer to each other.
+
+> *rec-attrset* = `rec {` [ *name* `=` *expr* `;` `]`... `}`
```

Comment:
```suggestion
> `rec {` [ *name* `=` *expr* `;` `]`... `}`
```

`*rec-attrset*` is not part of the syntax for construcing an attrset.

Reply:
I was loosely following EBNF by of assigning symbolic names to syntactic constructs, see [here](https://github.com/NixOS/nix/pull/8317/files#diff-5c02769fe3c7f1713efed6441672ef342bb21f7a9af62b572d7c3db2b20cc130R170-R171) for example.

That change would make it inconsistent. Or are you suggesting this is in fact an `attrset` and we should have an optional ``[`rec`]``? That would mess up the setup quite a bit...

While slowly chipping at it I got a strong urge to fix https://github.com/NixOS/nix/issues/7259, but this would be a lot more work. I imagine it to be easier to just collect bits and pieces and then compile them into a whole when it looks good enough.

Reply:
@edolstra what do you think?